### PR TITLE
Docker file.

### PR DESCRIPTION
**Pull request by alexggmatthews**
Implements #335

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#USAGE:
+#To run from prebuilt version use:
+#docker run -it -p 8888:8888 gpflow/gpflow
+
+#To replicate build use:
+#docker build -t gpflow/gpflow .
+
+#Uses official Tensorflow docker for cpu only.
+FROM tensorflow/tensorflow
```
Comment (john-bradshaw) [inline Dockerfile]: You should choose a specific tag otherwise it will default to latest which may not be what you want to do?

Comment (alexggmatthews) [inline Dockerfile]: Seems sensible, thanks. Do you think we should also add use a specific version for our repository, or always assume we are using master? I guess we could do a couple of different images.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+RUN apt-get -y update
```
Comment (john-bradshaw) [inline Dockerfile]: you should merge these run steps (\S minimise the number of layers [1])

and add some cleaning after installing to keep the container small (\S apt-get [1]), ie something along the lines of:
```
 && apt-get clean  \
 && rm -rf /var/lib/apt/lists/* \
```

[1]. https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+RUN apt-get -y install git
+RUN git clone https://github.com/GPflow/GPflow /usr/local/GPflow
+RUN cd /usr/local/GPflow && \
+    python setup.py develop
+#Clear TensorFlow demos and replace them with GPflow examples.
+RUN rm /notebooks/*.*
+RUN rm /notebooks/*
```
Comment (john-bradshaw) [inline Dockerfile]: what does this achieve in addition to line 30?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+COPY doc/source/notebooks/*.* /notebooks/
```
Comment (john-bradshaw) [inline Dockerfile]: could you not copy from your cloned repo, rather from a one outside?

File: Dockerfile (hunk 1 above)
Comment (john-bradshaw) [inline Dockerfile]: yes I think that would be a good idea - match the docker tags with the github tags

**Comment by alexggmatthews** [comment]
@jameshensman we've had a good guest review from @john-bradshaw on this one. I've met all his comments. If you have anything to add let me know. Otherwise lets merge.

**Comment by jameshensman** [comment]
Lovely work @alexggmatthews , and thank for the input @john-bradshaw .

**Pull request merged**